Add -v/-vv CLI logging verbosity

armalikhani opened the pull request:
This PR adds `-v` / `-vv` flags to the `mqssbench` CLI to control logging verbosity.

- No `-v` → WARNING level logs (default)
- `-v` → INFO level logs
- `-vv` → DEBUG level logs

Changes included:
- Implement `-v` / `--verbose` as a repeatable flag
- Update README CLI Usage section with verbosity instructions and minor general improvements
- Update `examples/run_example.py` to match the latest result structure after the multi-config feature
  (Related to PR #23)

armalikhani committed:
✨ Add -v/-vv CLI logging verbosity

armalikhani committed:
📝 Update README with verbosity instructions and minor improvements

armalikhani committed:
🔧 Update example file to match latest result structure

mnfarooqi commented:
There are two ways to specify verbosity. One is '-v' and the other is '--verbose'. While '-v' and '-vv' are used for different verbosity levels, it would be better to have a verbosity level option that uses '--verbose', e.g. '--verbose=0'.

armalikhani committed:
✨ Switch CLI verbosity to numeric  --verbose=<n>

armalikhani commented:
Thanks for the review and suggestion. I updated the CLI to use a numeric verbosity flag via `--verbose=<n>` with the short alias `-v`, aligning with the suggested approach. I also updated the PR title, description, documentation, and examples accordingly.

mnfarooqi commented:
> Thanks for the review and suggestion. I updated the CLI to use a numeric verbosity flag via `--verbose=<n>` with the short alias `-v`, aligning with the suggested approach. I also updated the PR title, description, documentation, and examples accordingly.

Your previous implementation of the -v and -vv options for verbosity levels was fine. My comment was about the longer version, '--verbose'. I think you can use your previous approach. Just add to the help text that '--verbose' can be added multiple times to increase verbosity.

armalikhani commented:
Actually, in my implementation the short and long options are just aliases. After your comment, I reconsidered both, while repeatable `-v` is fast and convenient, numeric verbosity feel more explicit and intutive for user and scales better if we add more than three log levels in the future. If you agree, I suggest we proceed with this approach. (If you prefer supporting the repeatable form as well, I can add it, but I would recommend keeping a single, simple model.)

mnfarooqi commented:
> Actually, in my implementation the short and long options are just aliases. After your comment, I reconsidered both, while repeatable `-v` is fast and convenient, numeric verbosity feel more explicit and intutive for user and scales better if we add more than three log levels in the future. If you agree, I suggest we proceed with this approach. (If you prefer supporting the repeatable form as well, I can add it, but I would recommend keeping a single, simple model.)

If you want to use verbosity levels then I suggest to use "--verbosity" or "--log-level" instead of "-v" or "--verbose" as these are used without any level in other gnu tools.

armalikhani committed:
🔧 Switch back to repeatable -v/--verbose for CLI logging

Considering simplicity and ease of use, we decided to keep the repeatable -v approach.
Log-level support can be added in the future if needed.

armalikhani commented:
After our discussing, we agreed to revert to the repeatable `-v` / `--verbose` implementation.

Initially, I switched to the numeric `--verbose=<n>` approach as it might be more aligned with MQSS common practices and scalable in the long term, but since that wasn’t the case, and because numeric `--verbose` is less common, `--log-level` implies mapping to actual log levels (which is not needed yet), and `--verbosity` also feels less familiar, we decided against it.

Given this, the initial repeatable `-v` / `--verbose` approach is simpler and more convenient for users at this stage. Future improvements, such as more advanced log control, could add `--log-level` support if needed.
